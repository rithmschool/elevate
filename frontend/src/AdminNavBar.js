import React, { Component } from "react";


class AdminNavBar extends Component {
    handleClick = (evt) => {
      evt.preventDefault();
      this.props.changeView(evt.target.id);
    }

    render(){

    return (
      <div className="adminNavBar">
        <div id="users" onClick={this.handleClick}>Users</div>
        <div id="invoices" onClick={this.handleClick}>Invoices</div>
        <div id="templates" onClick={this.handleClick}>Templates</div>
        <div id="calendar" onClick={this.handleClick}>Calendar</div>
      </div>

    )
  }
}



export default AdminNavBar;