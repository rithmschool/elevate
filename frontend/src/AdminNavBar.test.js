import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from "enzyme-to-json";
import AdminNavBar from './AdminNavBar';

describe('AdminNavBar', function() {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<AdminNavBar />);
  });
  
  it('renders without crashing', function () {
    shallow(<AdminNavBar />);
  });
  
  it('matches snapshot', function () {
    let serialized = toJson(wrapper);

    expect(serialized).toMatchSnapshot();
  });

  it('has div with admin-navbar class', function () {
    expect(wrapper.find('div.adminNavBar')).toHaveLength(1);
  });

  it('has a div for users, invoices, templates, and calendars', function () {
    expect(wrapper.find('div[id="users"]')).toHaveLength(1);
    expect(wrapper.find('div[id="invoices"]')).toHaveLength(1);
    expect(wrapper.find('div[id="templates"]')).toHaveLength(1);
    expect(wrapper.find('div[id="calendar"]')).toHaveLength(1);
  });

  it('calls handleClick', function () {
    const spy = jest.spyOn(wrapper.instance(), "handleClick");
    wrapper.setProps({ changeView: function changeView(){} });
    wrapper.find('div[id="users"]').simulate('click');

    expect(spy).toHaveBeenCalled();
  });
});