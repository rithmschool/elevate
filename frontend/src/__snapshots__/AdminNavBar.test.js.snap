// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminNavBar matches snapshot 1`] = `
<AdminNavBar>
  <div
    className="adminNavBar"
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <div
      id="users"
      onClick={[Function]}
    >
       Users
    </div>
    <div
      id="questions"
      onClick={[Function]}
    >
      Questions
    </div>
    <div
      id="invoices"
      onClick={[Function]}
    >
      Invoices
    </div>
    <div
      id="calendar"
      onClick={[Function]}
    >
      Calendar
    </div>
  </div>
</AdminNavBar>
`;
